### PR TITLE
Instructor search

### PR DESCRIPTION
**Pull request by miriam-rittenberg**

### CODE REVIEW

**Comment by miriam-rittenberg** [comment]
Resolves #230

File: src/components/ClassSearch.vue [hunk 1]
```diff
@@ -237,30 +263,31 @@ class FilterGroup {
...
+var hassArt = new RegexFilter('HASS-A', 'A', ['hass_attribute'], 'HASS-A');
+var hassSocialScience = new RegexFilter('HASS-S', 'S', ['hass_attribute'], 'HASS-S');
+var hassHumanity = new RegexFilter('HASS-H', 'H', ['hass_attribute'], 'HASS-H');
+var ciAny = new RegexFilter('CI:Any', 'Any', ['communication_requirement'], 'CI.+');
+var ciH = new RegexFilter('CI-H', 'CI-H', ['communication_requirement'], 'CI-H');
+var ciHW = new RegexFilter('CI-HW', 'CI-HW', ['communication_requirement'], 'CI-HW');
+var ciNone = new RegexFilter('Not CI', 'None', ['communication_requirement'], '^(?!CI)');
+var levelUG = new RegexFilter('Undergraduate', 'UG', ['level'], 'U');
+var levelG = new RegexFilter('Graduate', 'G', ['level'], 'G');
 var unitsLt6 = new MathFilter('<6', '<6', [undefined, 6], false, ['total_units']);
```
Comment (npfoss) [inline src/components/ClassSearch.vue]: I'm 99% sure this one was also supposed to be changed, right?

File: src/components/ClassSearch.vue [hunk 2]
```diff
@@ -209,6 +209,32 @@ class BooleanFilter extends Filter {
...
+class ArrayFilter extends Filter {
+  constructor (name, shortName, attributeNames, SubfilterType, subfilterArguments, mode) {
+    var subfilter = new SubfilterType(name, shortName, attributeNames, ...subfilterArguments, mode);
+    var comparator = subfilter.filter;
+    super(name, shortName, comparator, attributeNames, mode);
+    this.subfilter = subfilter;
+  }
+  matches (subject) {
+    // starting value of true for and, false for or
+    var isMatch = !this.combine(true, false);
```
Comment (georgiashay) [inline src/components/ClassSearch.vue]: I am not sure that searching by instructor is worth the slowdown (probably caused by all these loops).  We should think about ways to improve the speed of this.  One thought i had is joining the instructors with commas into a single string when we pull the course catalog, then searching using a regex that avoids the comma boundaries.

**Comment by npfoss** [comment]
(this is also going to conflict with #393 so might not be worth resolving conflicts until then anyways, so you can also add the new one to the unit tests? sorry :P )

**Pull request closed without merging**